gems/sorbet-runtime/lib/types/private/methods/call_validation.rb

```diff
@@ -214,6 +214,7 @@ def self.create_validator_method_fast(mod, original_method, method_sig)
   end
 
   def self.create_validator_method_fast0(mod, original_method, method_sig, return_type)
+    T::Private::DeclState.current.is_re_wrapping = true
```

Comment:
This allows us to know that the call to `_on_method_added` that's about to happen is merely a result of the fact that we do some of the sig-wrapping logic lazily, and we therefore shouldn't do the final checks right now. The way we communicate this fact, however, is via global mutable state, which is generally not good.